pkg/gatherers/conditional/gather_pod_definition.go

```diff
@@ -0,0 +1,91 @@
+package conditional
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/insights-operator/pkg/gatherers"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// BuildGatherPodDefinition collects pod definition from pods that are firing one of the configured alerts.
+//
+// * Location in archive: conditional/namespaces/<namespace>/pods/<pod>.json
+// * Id in config: pod_definition
```

Comment:
Hmm do we use these ids for the conditional gathering at all?

Reply:
Not that I'm aware. Maybe with conditional service running? We do have it in others CG if I'm not wrong...

Reply:
It's used only in the `GatheringFunctionName` (e.g https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/conditional/gathering_functions.go#L19), but I am not sure if it's taken into account in the https://github.com/openshift/insights-operator/blob/master/pkg/gather/gather.go#L203. It seems it is. It would be good to verify and then unify across all the conditional gatherer. And yes it's a good question how it will work with the new service